compat/yoast-sitemap.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * Returns a list of all images added using Page Builder.
+ *
+ * @param $images an array of all detected images used in the current post.
+ * @param $post_id the current post id.
+ *
+ * @return array
+ */
+function siteorigin_yoast_sitemap_images_compat( $images, $post_id ) {
+	if ( get_post_meta( $post_id, 'panels_data', true ) && class_exists( 'DOMDocument' ) ) {
+		$content = SiteOrigin_Panels::renderer()->render(
+			$post_id,
+			false
+		);
+
+		libxml_use_internal_errors( true );
+		$dom = new DOMDocument();
```

Comment:
If a user doesn't have `php-xml` installed, then this will cause a fatal error. I don't think it's very common, but I remember encountering this back in my early days of theme dev. It's worth throwing in a `class_exists( 'DOMDocument' )` check at the start of `siteorigin_yoast_sitemap_images_compat`.

That said, maybe Yoast is already checking for this, so a user will never hit the `wpseo_sitemap_urlimages` filter if they don't have `php-xml` installed.

Reply:
Yoast checks if `DOMDocument` is enabled (if it's not, it'll never trigger our code), but they're not checking if `php-xml` is. Their usage is significantly before ours so if there is a fatal error, it'll be with their code first. It would be considering adding a `php-xml` check in the offchance they add a check also.

Is your preference with checking with `extension_loaded` or will `function_exists` suffice?

Reply:
Either would be good here.